### PR TITLE
Update tools to fix CI

### PR DESCRIPTION
**Pull request by lmolkova**

### CODE REVIEW

File: semantic-conventions/src/opentelemetry/semconv/model/utils.py
Comment (lmolkova) [inline semantic-conventions/src/opentelemetry/semconv/model/utils.py:16]: used implicitly, but flake does not understand it

**Review by trask — APPROVED** [review]

**Comment by trask** [comment]
@open-telemetry/technical-committee @open-telemetry/specs-semconv-maintainers can you review / merge this to fix the build (and avoid this being a blocker for #230)? thx

**Review by AlexanderWert — APPROVED** [review]

File: semantic-conventions/src/opentelemetry/semconv/model/utils.py
Comment (Oberon00) [reply semantic-conventions/src/opentelemetry/semconv/model/utils.py:16]: I think the "proper" solution would be to switch from legacy `#type:` comments to using actual annotations.

**Review by Oberon00 — APPROVED** [review]

**Comment by Oberon00** [comment]
We could also pin the python minor version when calling the setup-python action to avoid being blocked by such things (although it is nice to update the tools once in a while).

File: semantic-conventions/src/opentelemetry/semconv/model/utils.py
Comment (arminru) [reply semantic-conventions/src/opentelemetry/semconv/model/utils.py:16]: I created https://github.com/open-telemetry/build-tools/issues/234 as a follow-up but will merge this PR as-is for now.
cc @lmolkova

**Review by arminru — APPROVED** [review]

**Pull request merged**
